Sync stackhpc/2024.1 into master

Alex-Welsh opened the pull request.

technowhizz committed:
Convert smartmon script to python

technowhizz committed:
Create tests for smartmon

technowhizz committed:
Use pySMART

technowhizz committed:
Add tests for pysmart

priteau committed:
Fix "Max Inlet Temp" time series chart

This chart could wrongly display CPU2 temperature instead of inlet
temperature.

Co-authored-by: Will Szumski <will@stackhpc.com>

Alex-Welsh committed:
Pin remaining Ansible dependencies (#1495)

Pin remaining Ansible dependencies

assumptionsandg committed:
Kolla dependency updates on non-default branches (#1552)

* Allow kolla tag updates on non-default branches

* Fixup matrices

* Reno

* Include version in title

assumptionsandg committed:
Bump kayobe to stackhpc/16.5.0.2

assumptionsandg committed:
Bump kolla to stackhpc/18.5.0.2

Alex-Welsh committed:
Merge pull request #1544 from stackhpc/redfish-dashboard-fix-inlet-temp

Fix "Max Inlet Temp" time series chart

technowhizz committed:
Fix broken links & ensure all kayobe, kolla & KA docs links use extlinks (#1557)

Fix broken upstream doc links

Alex-Welsh committed:
Merge pull request #1556 from stackhpc/update-dependency/kolla/stackhpc/2024.1

Bump stackhpc/2024.1 kolla to stackhpc/18.5.0.2

Alex-Welsh committed:
Merge pull request #1555 from stackhpc/update-dependency/kayobe/stackhpc/2024.1

Bump stackhpc/2024.1 kayobe to stackhpc/16.5.0.2

assumptionsandg committed:
Bump kolla-ansible to stackhpc/18.5.0.2

Alex-Welsh committed:
Merge pull request #1554 from stackhpc/update-dependency/kolla-ansible/stackhpc/2024.1

Bump stackhpc/2024.1 kolla-ansible to stackhpc/18.5.0.2

jackhodgkiss committed:
feat: use `include_tags` for kolla repositories

assumptionsandg committed:
Fix brackets in fluentd opensearch config

priteau committed:
Merge pull request #1560 from stackhpc/pulp-include-tags

feat: use `include_tags` for kolla repositories

assumptionsandg committed:
Fix release note

priteau committed:
Merge pull request #1571 from stackhpc/fix-fluentd-template

Fix brackets in fluentd opensearch config

grzegorzkoper committed:
Filter out interfaces that don't matter for packet drops

OVS interfaces drop packets during normal operation, generating lots of spurious alerts.

MoteHue committed:
Fix ceph-rgw blackbox endpoint enabling

When using the Ceph ingress for balancing RGW traffic, we disable the KA
loadbalancing of this traffic. Therefore, we do not need to monitor the
``ceph-rgw`` endpoints when ``enable_ceph_rgw_loadbalancer=false``.

MoteHue committed:
Update heat removal docs

We need to also delete the ``heat-cfn`` service. The endpoint deletions
are redundant, as endpoints are removed when the services are deleted.

priteau committed:
Merge pull request #1575 from stackhpc/docs-remove-heat-cfn-endpoint

Update heat removal docs

MoteHue committed:
Add a confirmation prompt to reboot.yml

This is a risky playbook to run without confirmation, especially as it
targets all hosts when not using a limit.

seunghun1ee committed:
Docs from openstack admin guide

(cherry picked from commit 80f8cdfd70c9b436b1d9e0e1f91b080a9891278f)

grzegorzkoper committed:
Merge branch 'stackhpc/2024.1' into fix-ovs-spurious-alerts

MoteHue committed:
Docs: add warning about Ceph when updating packages

jovial committed:
Merge pull request #1577 from stackhpc/docs-from-openstack-admin-guide-caracal

Docs from openstack admin guide

grzegorzkoper committed:
Merge pull request #1572 from stackhpc/fix-ovs-spurious-alerts

Filter out interfaces that don't matter for packet drops.

technowhizz committed:
Update docstring to state expected Device

technowhizz committed:
Add a function for converting to camelcase

MoteHue committed:
Merge pull request #1574 from stackhpc/fix-ceph-rgw-lb-blackbox-endpoint

Fix ceph-rgw blackbox endpoint enabling

darmach committed:
[CI] Fixes update Kolla versions running on clones

Recently added 'Update Kolla versions' job was missing conditional prohibiting
execution in non-StackHPC repository, i.e. customer fork.

technowhizz committed:
Split device and attribute tests into individual test cases per fixture for better error reporting

technowhizz committed:
Use function for snake case conversion

technowhizz committed:
Nit: Add .git to all GitHub repos for consistency

bbezak committed:
Bump kolla to stackhpc/18.6.0.2

bbezak committed:
Merge pull request #1581 from stackhpc/darmach-update-kolla-versions-conditional

[CI] Fixes update Kolla versions running on clones

bbezak committed:
Merge branch 'stackhpc/2024.1' into update-dependency/kolla/stackhpc/2024.1

bbezak committed:
Merge pull request #1568 from stackhpc/update-dependency/kolla/stackhpc/2024.1

Bump stackhpc/2024.1 kolla to stackhpc/18.6.0.2

priteau committed:
Merge pull request #1578 from stackhpc/ceph-package-update-warning

Docs: add warning about Ceph when updating packages

MoteHue committed:
Fix download of IPA images from Ark in Bifrost

MoteHue committed:
Update releasenotes/notes/add-confirmation-prompt-to-reboot.yml-4fd1ae8e8d360e57.yaml

Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

MoteHue committed:
Update etc/kayobe/ansible/reboot.yml

MoteHue committed:
Update releasenotes/notes/add-confirmation-prompt-to-reboot.yml-4fd1ae8e8d360e57.yaml

MoteHue committed:
change yes to true

MoteHue committed:
Merge pull request #1584 from stackhpc/bifrost-ipa-authenticated-download

Fix download of IPA images from Ark in Bifrost

assumptionsandg committed:
Bump kayobe to stackhpc/16.5.0.5

assumptionsandg committed:
Bump kolla-ansible to stackhpc/18.6.0.1

Alex-Welsh committed:
Merge pull request #1569 from stackhpc/update-dependency/kolla-ansible/stackhpc/2024.1

Bump stackhpc/2024.1 kolla-ansible to stackhpc/18.6.0.1

Alex-Welsh committed:
Merge pull request #1567 from stackhpc/update-dependency/kayobe/stackhpc/2024.1

Bump stackhpc/2024.1 kayobe to stackhpc/16.5.0.5

priteau committed:
Merge pull request #1583 from stackhpc/technowhizz-patch-1

Nit: Add .git to all Github Repos in kolla.yml

Alex-Welsh committed:
Add defaults for GPU PCI passthrough configuration

assumptionsandg committed:
Fix Kolla dependencies update schedule (#1585)

* Run weekly instead of nightly

* Include CI labels

* Fix ident

* Include changelog

* Remove version

* Change commit author to stackhpc-ci

priteau committed:
Fix typo in octavia-amphora-image-register playbook

priteau committed:
Merge pull request #1587 from stackhpc/typo-fix

Fix typo in octavia-amphora-image-register playbook

stackhpc-ci committed:
Bump kayobe to stackhpc/16.6.0.1

stackhpc-ci committed:
Bump kolla-ansible to stackhpc/18.6.0.2

assumptionsandg committed:
Replace 'stackhpc-ci' label with 'automated' (#1593)

Alex-Welsh committed:
Merge pull request #1592 from stackhpc/update-dependency/kolla-ansible/stackhpc/2024.1

Bump kolla-ansible to stackhpc/18.6.0.2

Alex-Welsh committed:
Merge pull request #1591 from stackhpc/update-dependency/kayobe/stackhpc/2024.1

Bump kayobe to stackhpc/16.6.0.1

stackhpc-ci committed:
Bump kolla to stackhpc/18.6.0.3

Alex-Welsh committed:
Merge pull request #1589 from stackhpc/update-dependency/kolla/stackhpc/2024.1

Bump kolla to stackhpc/18.6.0.3

seunghun1ee committed:
Add noble configs to Caracal

seunghun1ee committed:
Add Noble pulp repo versions for Caracal

Also updated UCA and Jammy repos

seunghun1ee committed:
Create caracal-noble workflows

Alex-Welsh committed:
Merge pull request #1576 from stackhpc/add-confirmation-to-reboot

Add a confirmation prompt to reboot.yml

seunghun1ee committed:
Remove unnecessary UCA repo for Caracal Noble

seunghun1ee committed:
Split Jammy and Noble case in kolla override

seunghun1ee committed:
Match Noble base tag to what stackhpc/master has

seunghun1ee committed:
Add Caracal Noble overcloud host image

seunghun1ee committed:
Add Caracal Noble IPA images

seunghun1ee committed:
Remove .git for glance_store override to fix pip dependency conflict

seunghun1ee committed:
Only enable Ubuntu repos match current dist release

seunghun1ee committed:
Add TODO for testing Ubuntu Noble CIS rules

seunghun1ee committed:
Add Noble CI-AIO tests for PRs

priteau committed:
Merge pull request #1596 from stackhpc/fix-glance-build

Remove .git for glance_store override to fix pip dependency conflict

seunghun1ee committed:
Add Caracal Noble Kolla container images

seunghun1ee committed:
Override K-A python version and base distro version for noble

jackhodgkiss committed:
feat: enable `skyline` build

seunghun1ee committed:
Bump sct_docker_version_max to 28.1.0

Default version of docker running on Ubuntu Noble is 28.0.4.
This version is now tested.

As SCT is using this value as exlusive max, bumping to 28.1.0 to
ensure 28.0.4 is included.

MoteHue committed:
Fix ssh in Ubuntu overcloud host images

Bumps the ``stackhpc_ubuntu_jammy_overcloud_host_image_version`` to fix
an issue where ``authorized_keys`` for the ``ubuntu`` user was not
populated on newly provisioned nodes.

seunghun1ee committed:
Fix kolla image build workflow for 2024.1

Kolla upstream change https://review.opendev.org/c/openstack/kolla/+/932386
brought Ubuntu 24.04 support for Caracal and split soruces.list.ubuntu
to sources.list.ubuntu.jammy and sources.list.ubuntu.noble to cover each
Ubuntu release.

MoteHue committed:
Bump Ubuntu pulp-repo-versions

jackhodgkiss committed:
feat: disable `ironic_prometheus_exporter`

assumptionsandg committed:
Beokay usage documentation

assumptionsandg committed:
Update doc/source/usage.rst

Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

assumptionsandg committed:
Update doc/source/usage.rst

Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

assumptionsandg committed:
Update doc/source/usage.rst

Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

Alex-Welsh committed:
Merge pull request #1605 from stackhpc/add-beokay-doc

Beokay usage documentation

technowhizz committed:
Move fixtures to tests folder

Alex-Welsh committed:
Work around SElinux upgrade issue

seunghun1ee committed:
Merge pull request #1604 from stackhpc/fix-selinux-ci

Fix selinux error in upgrade jobs

Alex-Welsh committed:
Merge branch 'stackhpc/2024.1' into fix-ssh-ubuntu-host-image

jackhodgkiss committed:
Merge branch 'stackhpc/2024.1' into enable-skyline-build

Alex-Welsh committed:
Merge pull request #1601 from stackhpc/fix-ssh-ubuntu-host-image

Fix ssh in Ubuntu overcloud host images

Alex-Welsh committed:
Apply suggestions from code review

Co-authored-by: Matt Crees <mattc@stackhpc.com>

seunghun1ee committed:
Support Jammy to Noble upgrade

Alex-Welsh committed:
Merge pull request #1603 from stackhpc/disable-ironic-prometheus

feat: disable `ironic_prometheus_exporter`

jackhodgkiss committed:
Merge pull request #1597 from stackhpc/enable-skyline-build

feat: enable `skyline` build

stackhpc-ci committed:
Bump kolla to stackhpc/18.6.0.5

stackhpc-ci committed:
Bump kolla-ansible to stackhpc/18.6.0.5

stackhpc-ci committed:
Bump kayobe to stackhpc/16.6.0.2

seunghun1ee committed:
Capitalisation

Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

seunghun1ee committed:
Add Ubuntu Noble support release note

Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

Alex-Welsh committed:
Merge pull request #1608 from stackhpc/update-dependency/kayobe/stackhpc/2024.1

Bump kayobe to stackhpc/16.6.0.2

Alex-Welsh committed:
Merge pull request #1606 from stackhpc/update-dependency/kolla/stackhpc/2024.1

Bump kolla to stackhpc/18.6.0.5

Alex-Welsh committed:
Merge pull request #1599 from stackhpc/update-dependency/kolla-ansible/stackhpc/2024.1

Bump kolla-ansible to stackhpc/18.6.0.5

seunghun1ee committed:
Add skyline image for Ubuntu Noble

jasonrayne committed:
Fix playbook failure due to invalid escaped character in service restart command

The playbook was failing when attempting to restart OpenStack services
because of an invalid escaped character in the command that lists and
restarts services. The original implementation used a shorthand
systemctl flag (-a) along with a regular expression for egrep, which led
to improper parsing and execution issues during the playbook run.

This commit replaces the shorthand with a more explicit invocation that
is confirmed to run without error:
- Uses "systemctl list-units --type=service --all --no-legend --plain"
to reliably list service units.
- Adds the "-r" flag to xargs to avoid running systemctl restart without
input, ensuring more predictable behavior.

Note: This bug may be specific to hosts running Ubuntu Jammy

Alex-Welsh committed:
Merge pull request #1609 from jasonrayne/stackhpc/2024.1

Fix playbook failure due to invalid escaped character

Alex-Welsh committed:
Merge pull request #1602 from stackhpc/fix-2024.1-kolla-image-build

Fix kolla image build workflow for 2024.1

m-bull committed:
Fix credential existence check for rgw usage exporter

credential_check.stdout is a string representation of a list, and as such will never return true when compared to an empty Python list.

Passing credential_check.stdout through the from_json filter casts the string to a list, and the equality can be properly evaluated.

m-bull committed:
Use system trust store as REQUESTS_CA_BUNDLE

By default Python requests uses its own CA bundle rather
than the system trust store. As we optionally update
the system trust store with our own CA, force Python
requests to use the system trust store as its CA bundle.

m-bull committed:
Merge pull request #1610 from stackhpc/fix/rgw-usage-exporter-creds-check

Fix RADOS gateway usage exporter deployment

Alex-Welsh committed:
Merge pull request #1586 from stackhpc/pci-passthrough-defaults

Add defaults for GPU PCI passthrough

stackhpc-ci committed:
Bump kayobe to stackhpc/16.6.0.3

stackhpc-ci committed:
Bump kolla-ansible to stackhpc/18.6.0.6

Alex-Welsh committed:
Merge remote-tracking branch 'origin/update-dependency/kayobe/stackhpc/2024.1' into update-dependency/kolla-ansible/stackhpc/2024.1

jackhodgkiss committed:
feat: add `alertmanager-0.28.1`

MoteHue committed:
Add note on unsealing the Vault

Alex-Welsh committed:
Merge pull request #1615 from stackhpc/note-to-unseal-vaults

Add note on unsealing the Vault

Alex-Welsh committed:
Fix Ceph release conditional

Alex-Welsh committed:
Merge branch 'stackhpc/2024.1' into sync-2024.1

Alex-Welsh committed:
Merge pull request #1614 from stackhpc/update-dependency/kolla-ansible/stackhpc/2024.1

Bump kolla-ansible and kayobe versions

jackhodgkiss committed:
Merge branch 'stackhpc/2024.1' into alertmanager-0.28.1

priteau committed:
Merge pull request #1616 from stackhpc/ceph-facts

Fix Ceph release conditional

Alex-Welsh committed:
Bump ansible-core version in lint jobs

MoteHue committed:
Add script to migrate RabbitMQ queues

Include custom text colours to make it clear which outputs are good and
which are bad.

MoteHue committed:
Merge pull request #1619 from stackhpc/caracal-rabbitmq-queue-migration

Add script to migrate RabbitMQ queues

jackhodgkiss committed:
feat: add support for `TLS` generation with `OpenBao`

oneswig committed:
Fix typo in OFED guide

entires -> entries

Alex-Welsh committed:
Merge pull request #1621 from stackhpc/fix-ofed-typo

Fix typo in OFED guide

seunghun1ee commented on etc/kayobe/kolla.yml:
```diff
@@ -285,7 +285,7 @@ kolla_build_blocks:
   base_ubuntu_package_sources_list: |
     {% if stackhpc_kolla_clean_up_repo_mirrors | bool %}
-    COPY sources.list.ubuntu /etc/apt/sources.list.backup
+    COPY sources.list.ubuntu.noble /etc/apt/sources.list.backup
```
https://review.opendev.org/c/openstack/kolla/+/907589 shows It's just ``sources.list.ubuntu`` for master branch. Probably because it's just noble.

Alex-Welsh replied:
Done

Alex-Welsh committed:
Fix sources.list

Alex-Welsh committed:
Fix kolla image build tagging and build Skyline

jackhodgkiss committed:
Update requirements.yml

Co-authored-by: Will Szumski <will@stackhpc.com>

jackhodgkiss committed:
Merge branch 'stackhpc/2024.1' into add-openbao

jackhodgkiss committed:
feat: `lint` and `tox` run before `AIO` tests

Avoid running more resource intensive tests before linting and
unit tests. As subsequent patches to fix linting and tox issues will
require running AIOs again.

Alex-Welsh committed:
Merge pull request #1623 from stackhpc/perform-linting-first

feat: `lint` and `tox` run before `AIO` tests

Alex-Welsh commented:
Images built with the wrong tag, I'll fix the workflow and rebuild
`master-Object-noble-20250424T154342`

jackhodgkiss committed:
feat: add deprecation for `vault` in `TLS`

jackhodgkiss committed:
fix: add `# noqa: fqcn` to avoid errors with hashivault modules

jackhodgkiss committed:
Merge branch 'stackhpc/2024.1' into add-openbao

jackhodgkiss committed:
Merge branch 'stackhpc/2024.1' into alertmanager-0.28.1

jackhodgkiss committed:
fix: handle undefined variable in `msteams` example

jackhodgkiss committed:
Merge pull request #1617 from stackhpc/alertmanager-0.28.1

stackhpc-ci committed:
Bump kolla to stackhpc/18.6.0.7

stackhpc-ci committed:
Bump kolla-ansible to stackhpc/18.6.0.7

stackhpc-ci committed:
Bump kayobe to stackhpc/16.6.0.4

Alex-Welsh committed:
Merge pull request #1628 from stackhpc/update-dependency/kayobe/stackhpc/2024.1

Bump kayobe to stackhpc/16.6.0.4

Alex-Welsh committed:
Merge pull request #1625 from stackhpc/update-dependency/kolla/stackhpc/2024.1

Bump kolla to stackhpc/18.6.0.7

Alex-Welsh committed:
Merge pull request #1627 from stackhpc/update-dependency/kolla-ansible/stackhpc/2024.1

Bump kolla-ansible to stackhpc/18.6.0.7

Alex-Welsh commented:
Depends on https://github.com/stackhpc/stackhpc-kayobe-config/pull/1620

MoteHue committed:
Add upgrade-prerequisites.sh

MoteHue committed:
Merge pull request #1629 from stackhpc/add-upgrade-prereqs

Add upgrade-prerequisites.sh

Alex-Welsh committed:
Merge branch 'stackhpc/master' into sync-2024.1

seunghun1ee committed:
Add documentation for Ubuntu Noble upgrade

seunghun1ee committed:
Merge pull request #1624 from stackhpc/jammy-to-noble

Support Jammy to Noble upgrade

technowhizz committed:
Use prometheus_client for writing out metrics

technowhizz committed:
Add args and returns to doc string for parse_ifattributes

technowhizz committed:
Ensure metric names start with smartmon

technowhizz committed:
Update smartmon playbook for smartmon.py

technowhizz committed:
Add python script to generate test fixtures from real drives

technowhizz committed:
Remove smartmon.sh

technowhizz committed:
Add test fixture for Dell ENT NVMe

jovial committed:
Redfish Exporter v2.x (#1549)

* Configuration for the StackHPC fork of Redfish Exporter

* Update redfish dashboard

- Compatability with Dell servers
- Added health round up panels
- Add variables for different groups e.g overcloud vs compute

* Increase query lookback delta

Prometheus will regard metrics collected over this perioid as stale as
such they won't show up in grafana.

* Add raw tags

* Re-tweak Redfish dashboard for Lenovo

* Restore old UUID of the redfish dashboard

The redfish dashboard was updated to use a new UUID, however this caused
errors in the Grafana logs and Grafana didn't update the dashboard.
Reverting this to the old UUID fixes the issue.

* Make query lookback delta setting smarter

* Actually set lookback delta

* Adds openstack exporter scrape interval

* Add some raw tags

* Slightly less offensive variant on raw tags

* Refresh release note

* Update dashboard

* Apply suggestions from code review

Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

* Bump to redfish exporter v2.1.1-stackhpc

* Temporarily remove redfish log collection

This isn't working yet (takes far too long to scrape),let's
split it out into a separate patch.

Also removed the query lookback delta config, as this was
working around the much longer scrape times.

---------

Co-authored-by: Matt Anson <matta@stackhpc.com>
Co-authored-by: technowhizz <7688823+technowhizz@users.noreply.github.com>
Co-authored-by: Matt Crees <mattc@stackhpc.com>
Co-authored-by: Alex-Welsh <112560678+Alex-Welsh@users.noreply.github.com>

technowhizz committed:
Add conditional https to rgw docs

stackhpc-ci committed:
Bump kolla-ansible to stackhpc/18.6.0.8

Alex-Welsh committed:
Merge pull request #1634 from stackhpc/ceph-rgw-docs-update-0525

Add conditional https to rgw docs

Alex-Welsh committed:
Merge branch 'stackhpc/2024.1' into update-dependency/kolla-ansible/stackhpc/2024.1

Alex-Welsh committed:
Merge pull request #1637 from stackhpc/update-dependency/kolla-ansible/stackhpc/2024.1

Bump kolla-ansible to stackhpc/18.6.0.8

jackhodgkiss committed:
Merge pull request #1622 from stackhpc/add-openbao

feat: add support for `TLS` generation with `OpenBao`

technowhizz committed:
Merge pull request #1327 from stackhpc/smartmon-py

INFRA-388 Converting smartmon into python and adding mock tests

Alex-Welsh committed:
Merge branch 'stackhpc/master' into sync-2024.1

Alex-Welsh committed:
Merge remote-tracking branch 'origin/stackhpc/2024.1' into sync-2024.1

Alex-Welsh committed:
Merge branch 'stackhpc/master' into sync-2024.1

Alex-Welsh committed:
Add Skyline image tags
